tests/test_schema.py

```diff
@@ -731,6 +783,50 @@ def test_function_custom_enum_default_value():
     assert function_custom_enum_default_value.tags == []
 
 
+@GPTEnabled
+def function_custom_enum_list(
+    a: int,
+    b: str,
+    c: bool = False,
+    d: list[CustomEnum] = [CustomEnum.A, CustomEnum.B],
+):
+    """
+    This is a test function.
+
+    :param a: This is a parameter
+    :param b: This is another parameter
+    :param c: This is a boolean parameter
+    :param d: This is a list parameter
+    """
+    return a, b, c, d
+
+
+def test_function_custom_enum_list():
+    rf = ReferenceSchema(function_custom_enum_list)
+    b = rf.get_param("d")
+    b["default"] = ["A", "B"]
+    b["items"] = {"type": "string", "enum": ["A", "B", "C"]}
+
+    assert function_custom_enum_list.schema.to_json() == rf.schema
+    assert function_custom_enum_list.tags == []
+
+    # Verify we can invoke the function providing the encoded enum value
+    _, _, _, d = function_custom_enum_list(1, "", False, ["A"])
+    assert d == [CustomEnum.A]
```

Comment:
Another one I would add is an empty list.

Reply:
Should be done with 3354a3897cce25fc08083c4d6f2169987d53418b